2023-09-21-21:25:42   [info]: WandB initialized
project_name: HumanPoseForecasting, entity: None, group: None, job_type: None, resume: None, mode: online
2023-09-21-21:25:42   [config]:
  training_scheme: auto_regressive
  random_seed: 42
  batch_size: 32
  num_epochs: 100
  num_train_iterations: 1000
  num_eval_iterations: 200
  seq_length: 10
  checkpoint_frequency: 5
  model: {'type': 'st_transformer', 'num_blocks': 8, 'input_dropout': 0.1, 'embedding_dim': 64, 'positional_encoding': {'type': 'sin'}, 'transformer': {'type': 'parallel', 'spatial_heads': 4, 'temporal_heads': 4, 'spatial_dropout': 0.1, 'temporal_dropout': 0.1, 'ff_dimension': 128, 'ff_dropout': 0.1, 'input_dropout': 0.1}}
  optimizer: {'type': 'Adam', 'learning_rate': 0.001, 'betas': [0.9, 0.98], 'epsilon': 1e-08}
  loss: {'type': 'position_mse', 'reduction': 'mean'}
  lr_scheduler: {'type': 'cosine', 'max_iterations': 100000, 'warmup': 2000, 'min_learning_rate': 0.0, 'update_frequency': 1000, 'base_lr': 0.0003}
  skeleton: {'type': 's21', 'num_joints': 21}
  joint_representation: {'type': 'pos', 'absolute': True, 'joint_dim': 3}
  evaluation: {'metrics': ['positional_mse', 'auc'], 'distribution_metrics': [], 'frequency': 1, 'exhaustive_evaluation': True, 'timesteps': [80, 120, 160, 200, 240, 280, 320]}
  dataset: {'name': 'h36m', 'seed_length': 10, 'target_length': 10, 'downsampling_factor': 2, 'spacing': 5, 'training_size': 1000, 'normalize_orientation': False, 'validation_size': 1000, 'test_size': 1000, 'reverse_prob': 0.5}
  data_augmentation: {'normalize': True, 'reverse_prob': 0.0, 'snp_noise_prob': 0.0, 'snp_noise_portion': [0.05, 0.4], 'joint_cutout_prob': 0.0, 'joint_cutout_portion': [1, 4], 'timestep_cutout_prob': 0.0, 'timestep_cutout_portion': [1, 4], 'gaussian_noise_prob': 0.0, 'gaussian_noise_std': 0.005}2023-09-21-21:25:42   [info]: Initialized trainer for run: representation_study/position_abs_1
2023-09-21-21:25:42   [info]: Using device: cuda
2023-09-21-21:25:42   [info]: Training Queue:representation_study/baseline_1
 representation_study/baseline_geo_1
 representation_study/position_abs_1
2023-09-21-21:25:42   [info]: Calling: initialize_model...
2023-09-21-21:25:42   [info]: Initialized model
2023-09-21-21:25:42   [info]: Calling: initialize_optimization...
2023-09-21-21:25:42   [info]: Initialized optimizer
2023-09-21-21:25:42   [info]: Calling: load_train_data...
2023-09-21-21:25:54   [info]: Loaded training data: Length: 77447, Batched length: 2420, Iterations per epoch: 1000
2023-09-21-21:25:54   [info]: Calling: initialize_evaluation...
2023-09-21-21:25:54   [info]: Load the evaluation data for each action
2023-09-21-21:25:58   [info]: Initialized an evaluation for joint distances with 200
2023-09-21-21:25:58   [info]: Calling: train...
2023-09-21-21:25:58   [info]: Calling: training_loop...
2023-09-21-21:25:58   [info]: Start training for run representation_study/position_abs_1
2023-09-21-21:25:58   [info]: Calling: evaluate...
2023-09-21-21:25:58   [info]: Start evaluation on H3.6M dataset using actions: ['directions', 'discussion', 'eating', 'greeting', 'posing', 'phoning', 'purchases', 'sitting', 'sittingdown', 'smoking', 'takingphoto', 'waiting', 'walking', 'walkingdog', 'walkingtogether']
2023-09-21-21:25:58   [error]: Traceback (most recent call last):
  File "/home/user/denninge/human-pose-forecasting/src/utils/logging.py", line 39, in try_call_log
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/denninge/human-pose-forecasting/src/evaluation/EvaluationEngine.py", line 430, in evaluate
    self.evaluation_loop_distance(action, model, data_loader)
  File "/home/user/denninge/anaconda3/envs/vision_lab/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/denninge/human-pose-forecasting/src/evaluation/EvaluationEngine.py", line 466, in evaluation_loop_distance
    cur_input = self.data_augmentor(data[:, : self.seed_length], is_train=False)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/denninge/anaconda3/envs/vision_lab/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/denninge/human-pose-forecasting/src/data_utils/DataAugmentor.py", line 76, in forward
    return self.train_pipeline(x) if is_train else self.eval_pipeline(x)
                                                   ^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/denninge/human-pose-forecasting/src/data_utils/DataAugmentor.py", line 103, in <lambda>
    return lambda x: reduce(lambda acc, f: f(acc), funcs, x)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/denninge/human-pose-forecasting/src/data_utils/DataAugmentor.py", line 103, in <lambda>
    return lambda x: reduce(lambda acc, f: f(acc), funcs, x)
                                           ^^^^^^
  File "/home/user/denninge/human-pose-forecasting/src/data_utils/DataAugmentor.py", line 110, in _normalize
    return (x-self.norm_mean) / torch.sqrt(self.norm_var)
            ~^~~~~~~~~~~~~~~
RuntimeError: The size of tensor a (21) must match the size of tensor b (16) at non-singleton dimension 2

